packages/subproviders/src/subproviders/lattice.ts

```diff
@@ -0,0 +1,126 @@
+        };
+        this._latticeConnectClient = new config.latticeConnectClient(opts);
+    }
+
+    /**
+     * Fetches the current Lattice wallet Ethereum address at path `44'/60'/0'/0/0`. Only the 0-th index address
+     * may be fetched, but the Lattice user may switch wallets on their device at any time, in which case this
+     * function would return a new address (in the form of a 1-element string array).
+     * @param numberOfAccounts number of accounts to fetch. Currently this is ignored in the connect client, as only one address may be fetched at a time
+     * @return A one-element array of addresses representing the current Lattice wallet's Ethereum account.
+     */
+    public async getAccountsAsync(numberOfAccounts: number = DEFAULT_NUM_ADDRESSES_TO_FETCH): Promise<string[]> {
+        try {
+            const accounts = await this._latticeConnectClient.addAccounts(numberOfAccounts);
+            return accounts;
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    /**
+     * Signs a transaction from the account specified by the `from` field in txParams.
+     * @param txParams Parameters of the transaction to sign.
+     * @return Signed transaction hex string. This is a serialized `ethereum-tx` Transaction object.
+     */
+    public async signTransactionAsync(txData: PartialTxParams): Promise<string> {
+        if (txData.from === undefined || !addressUtils.isAddress(txData.from)) {
+            throw new Error(WalletSubproviderErrors.FromAddressMissingOrInvalid);
+        }
+        const txReq = new EthereumTx(txData);
+        try {
+            const signedTx = await this._latticeConnectClient.signTransaction(txData.from, txReq);
+            return `0x${signedTx.serialize().toString('hex')}`;
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    /**
+     * Sign a personal Ethereum message from the account specified in the `address` param.
+     * @param data Data to be signed. May be represented in hex or ASCII; this representation will be preserved.
+     * @param address Address from which to sign. Must be the address at `m/44'/60'/0'/0/0` of the current wallet.
+     * @return Signature hex string of form `0x{r}{s}{v}
+     */
+    public async signPersonalMessageAsync(data: string, address: string): Promise<string> {
+        if (data === undefined) {
+            throw new Error(WalletSubproviderErrors.DataMissingForSignPersonalMessage);
+        }
+        assert.isHexString('data', data);
+        assert.isETHAddressHex('address', address);
+        try {
+            const sig = await this._latticeConnectClient.signPersonalMessage(address, data);
+            return sig;
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    /**
+     * Sign a typed data message from the account specified in the `address` param.
+     * NOTE: This method is not implemented on the Lattice in v1, but we plan to support it in future versions.
```

Comment:
Might be best to throw an error whilst it's not implemented? Otherwise the dev may have a hard time tracking down a bug.

If it does throw, some tools will default to `eth_sign` of the hash of the data.

Reply:
So there is an error that gets thrown by the underlying SDK ([here](https://github.com/GridPlus/gridplus-sdk/blob/master/src/ethereum.js#L53)). That error will get sent up to the `_latticeConnectClient` ([here](https://github.com/GridPlus/eth-lattice-keyring/blob/master/index.js#L131) and [here](https://github.com/GridPlus/eth-lattice-keyring/blob/master/index.js#L152)) and will eventually make it to this module. We will be adding support in the next few months via the other two modules I linked (which we maintain). At that point, the `signMessage` functionality will be extensible to other signing schemas, should future protocols get ossified (since it will switch on that `protocol` value). 

Happy to remove it and come back, but I figured this would avoid having to go through two sequential PRs. But no problem either way!

Reply:
@dekz went ahead and removed the comment in ea71cfc

We are actually working on an implementation now. It may be available by the time we get an integration out that uses this module.